components/odh-notebook-controller/controllers/notebook_runtime.go

```diff
@@ -0,0 +1,244 @@
+	return true, nil // ConfigMap exists
+}
+
+func (r *OpenshiftNotebookReconciler) syncRuntimeImagesConfigMap(ctx context.Context, notebookNamespace string, controllerNamespace string, config *rest.Config) error {
+	log := r.Log.WithValues("namespace", notebookNamespace)
+
+	// Create a dynamic client
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		log.Error(err, "Error creating dynamic client")
+		return err
+	}
+
+	// Define GroupVersionResource for ImageStreams
+	ims := schema.GroupVersionResource{
+		Group:    "image.openshift.io",
+		Version:  "v1",
+		Resource: "imagestreams",
+	}
+
+	// Fetch ImageStreams from controllerNamespace namespace
+	imagestreams, err := dynamicClient.Resource(ims).Namespace(controllerNamespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		log.Error(err, "Failed to list ImageStreams", "Namespace", controllerNamespace)
+		return err
+	}
+
+	// Prepare data for ConfigMap
+	data := make(map[string]string)
+	for _, item := range imagestreams.Items {
+		labels := item.GetLabels()
+		if labels["opendatahub.io/runtime-image"] == "true" {
+			tags, found, err := unstructured.NestedSlice(item.Object, "spec", "tags")
+			if err != nil || !found {
+				log.Error(err, "Failed to extract tags from ImageStream", "ImageStream", item.GetName())
+				continue
+			}
+
+			for _, tag := range tags {
+				tagMap, ok := tag.(map[string]interface{})
+				if !ok {
+					continue
+				}
+
+				// Extract metadata annotation
+				annotations, found, err := unstructured.NestedMap(tagMap, "annotations")
+				if err != nil || !found {
+					annotations = map[string]interface{}{}
+				}
+
+				metadataRaw, ok := annotations["opendatahub.io/runtime-image-metadata"].(string)
+				if !ok || metadataRaw == "" {
+					metadataRaw = "[]"
+				}
+
+				// Parse metadata
+				metadataParsed := parseRuntimeImageMetadata(metadataRaw)
+				displayName := extractDisplayName(metadataParsed)
+
+				// Construct the key name
+				if displayName != "" {
+					formattedName := formatKeyName(displayName)
+					data[formattedName] = metadataParsed
+				}
+			}
+		}
+	}
+
+	// Create or update the ConfigMap in the Notebook's namespace
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configMapName,
+			Namespace: notebookNamespace,
+			Labels:    map[string]string{"opendatahub.io/managed-by": "workbenches"},
+		},
+		Data: data,
+	}
+
+	configMapExists, err := r.checkConfigMapExists(ctx, configMapName, notebookNamespace)
+	if err != nil {
+		log.Error(err, "Error checking if ConfigMap exists", "ConfigMap.Name", configMapName)
+		return err
+	}
+
+	if configMapExists {
+		existingConfigMap := &corev1.ConfigMap{}
+		if err := r.Get(ctx, types.NamespacedName{Name: configMapName, Namespace: notebookNamespace}, existingConfigMap); err != nil {
+			log.Error(err, "Failed to get existing ConfigMap", "ConfigMap.Name", configMapName)
+			return err
+		}
```

Comment:
seems like these line are repeating. Right ?
on line 115 , checkCongifMapExist function is used, 
can we restructure: `checkConfigMapExists` to `getRuntimeConfigMap`
such that `getRuntimeCOnfigMap` function, return 3 values (configmap, ifexits  bool, err) 
that would remove this 3 extra lines right ?

Reply:
Good catch! I updated and tested. Looks better now :)